### PR TITLE
Issue 17088 deleting duped workflow tasks

### PR DESCRIPTION
**Pull request by wezell**
De-dupes `workflow_tasks`

### CODE REVIEW

**Review by nollymar — COMMENTED** [review]
we should implement an IT

**Review by jgambarios — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Review by nollymar — COMMENTED** [review]
A new IT was implemented to test these changes

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java [hunk 1]
```diff
@@ -36,6 +41,37 @@ public void executeUpgrade() throws DotDataException, DotRuntimeException {
...
+            final List<String> deleteMes = dotConnect
+            .setSQL("select id from workflow_task where webasset=? and language_id=? order by mod_date desc")
+            .addParam(webAsset)
+            .addParam(lang)
+            .setStartRow(1) //magic happens here
+            .loadObjectResults()
+            .stream()
+            .map(r->(String)r.get("id")).collect(Collectors.toList());
+            
+            String subquery = "'" + String.join("','" , deleteMes) + "'";
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java]: Codacy Issue found: [Local variable 'subquery' could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4666366)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java [hunk 2]
```diff
@@ -7,7 +7,12 @@
 import com.dotmarketing.exception.DotRuntimeException;
 import com.dotmarketing.startup.StartupTask;
 import com.dotmarketing.util.Logger;
+import io.vavr.control.Try;
 import java.sql.SQLException;
+import java.util.Date;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Date'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4666366)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKeyTest.java [hunk 3]
```diff
@@ -49,4 +65,69 @@ public void Test_Upgrade_Task() throws DotDataException {
...
+        assertTrue(dupeWorkflowTask != null && dupeWorkflowTask.getId() != null);
+        assertEquals(workflowTask.getWebasset(), dupeWorkflowTask.getWebasset());
+        assertEquals(workflowTask.getLanguageId(), dupeWorkflowTask.getLanguageId());
+
+        final Task05200WorkflowTaskUniqueKey task =  new Task05200WorkflowTaskUniqueKey();
+        assertTrue(task.forceRun());
+        task.executeUpgrade();
+
+        //verify the first task and its dependencies have been deleted
+        final WorkflowTask result = workflowAPI.findTaskById(workflowTask.getId());
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKeyTest.java]: Codacy Issue found: [Avoid unused local variables such as 'result'.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4666366)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKeyTest.java [hunk 4]
```diff
@@ -49,4 +65,69 @@ public void Test_Upgrade_Task() throws DotDataException {
...
+        final WorkflowAPI workflowAPI = APILocator.getWorkflowAPI();
+
+        final Contentlet contentlet = TestDataUtils.getPageContent(true, language.getId());
+
+        //save workflow task
+        final WorkflowStep workflowStep           = workflowAPI.findStep(
+                SystemWorkflowConstants.WORKFLOW_NEW_STEP_ID);
+        workflowAPI.deleteWorkflowTaskByContentletIdAnyLanguage(contentlet, systemUser);
+        final WorkflowTask workflowTask = workflowAPI
+                .createWorkflowTask(contentlet, systemUser, workflowStep, "test", "test");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKeyTest.java]: Codacy Issue found: [The String literal "test" appears 4 times in this file; the first occurrence is on line 84](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4666366)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java [hunk 5]
```diff
@@ -36,6 +41,37 @@ public void executeUpgrade() throws DotDataException, DotRuntimeException {
             throw new DotDataException(e.getMessage(), e);
         }
         final DotConnect dotConnect = new DotConnect();
+        
+        List<Map<String,Object>> results = dotConnect.setSQL("select webasset,language_id from workflow_task group by webasset,language_id having count(*)>1").loadObjectResults();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05200WorkflowTaskUniqueKey.java]: Codacy Issue found: [Local variable 'results' could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4666366)

**Review by dsilvam — APPROVED** [review]

**Pull request merged**